### PR TITLE
MM-53813: Fix(accessibility): on channels page

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
- Fixed issues found after running [FastPass](https://accessibilityinsights.io/docs/web/getstarted/fastpass/) in the Channels page.
- Added Playwright accessibility test

Notes:
- Requires PR from dynamic-virtualized-list: https://github.com/mattermost/dynamic-virtualized-list/pull/28
- `color-contrast` ratio requirement to be addressed separately - https://mattermost.atlassian.net/browse/MM-53813

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-53813

#### Screenshots
Before:
Screenshot 2023-07-26 at 4 31 50 PM

Playwright test:
Screenshot 2023-07-26 at 7 06 20 AM

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by M-ZubairAhmed** [comment]
You are on 🔥 with these a11y fixes. Awesome

File: webapp/channels/package.json [hunk 1]
```diff
@@ -32,7 +32,7 @@
     "crypto-browserify": "3.12.0",
     "css-vars-ponyfill": "2.4.8",
     "date-fns": "2.29.3",
-    "dynamic-virtualized-list": "github:mattermost/dynamic-virtualized-list#1c330717d6778315a40e70137ace38247c6a1d0f",
+    "dynamic-virtualized-list": "github:mattermost/dynamic-virtualized-list#9695a52781981db77333607343ea62eac828bda4",
```
Comment (saturninoabril) [inline webapp/channels/package.json:35]: Will update once the related PR is merged.

**Review by yasserfaraazkhan — APPROVED** [review]
thank you for the changes @saturninoabril .

just a general thought 
I was wondering if there's away we can also capture in pr the before/ after or convert that PR meets A,AA level of success criteria through applied fixes

File: webapp/channels/src/components/__snapshots__/searchable_channel_list.test.tsx.snap [hunk 2]
```diff
@@ -65,7 +65,7 @@ exports[`components/SearchableChannelList should match init snapshot 1`] = `
   </div>
   <div
     className="more-modal__list"
-    role="application"
+    role="search"
```
Comment (saturninoabril) [inline webapp/channels/src/components/__snapshots__/searchable_channel_list.test.tsx.snap:68]: This fixes the issue in the screenshot.
Screenshot 2023-07-26 at 4 34 28 PM

File: webapp/channels/src/components/channel_header/__snapshots__/channel_header.test.tsx.snap [hunk 3]
```diff
@@ -4700,7 +4685,6 @@ exports[`components/ChannelHeader should render the pinned icon with the pinned
                     </span>
                   </strong>
                   <span
-                    aria-label="dropdown icon"
```
Comment (saturninoabril) [inline webapp/channels/src/components/channel_header/__snapshots__/channel_header.test.tsx.snap:4703]: This fixes the issue in the screenshot. Such ARIA label is redundant to the parent (button) ARIA label as `channel menu` which acts as single interactive element. Dropdown icon is not accessible/interactive by its own alone. 
Screenshot 2023-07-26 at 4 45 44 PM

File: webapp/channels/src/components/channel_header/channel_header.tsx [hunk 4]
```diff
@@ -785,7 +785,6 @@ class ChannelHeader extends React.PureComponent<Props, State> {
                             <span
                                 id='channelHeaderDropdownIcon'
                                 className='icon icon-chevron-down header-dropdown-chevron-icon'
-                                aria-label={formatMessage({id: 'generic_icons.dropdown', defaultMessage: 'Dropdown Icon'}).toLowerCase()}
```
Comment (saturninoabril) [inline webapp/channels/src/components/channel_header/channel_header.tsx:788]: This fixes the issue in the screenshot. Such ARIA label is redundant to the parent (button) ARIA label as `channel menu` which acts as single interactive element. Dropdown icon is not accessible/interactive by its own alone. 
Screenshot 2023-07-26 at 4 45 44 PM

File: webapp/channels/src/components/post/post_component.tsx [hunk 5]
```diff
@@ -513,7 +513,6 @@ const PostComponent = (props: Props): JSX.Element => {
             {(isSearchResultItem || (props.location !== Locations.CENTER && (props.isPinnedPosts || props.isFlaggedPosts))) && <DateSeparator date={currentPostDay}/>}
             <PostAriaLabelDiv
                 ref={postRef}
-                role='listitem'
```
Comment (saturninoabril) [inline webapp/channels/src/components/post/post_component.tsx:516]: This and the changes at [post_list_virtualized/post_list_virtualized.tsx](https://github.com/mattermost/mattermost/pull/24122/files#diff-61c4a17151b88abdf551ba2bc3a2298363473a02fd9352a8b878002fcd831069), and on [dynamic-virtualized-list](https://github.com/mattermost/dynamic-virtualized-list/pull/28) fixed the issue in the screenshot. It properly set the ARIA role's parent-children hierarchy in the post list content.
Screenshot 2023-07-26 at 4 52 14 PM

File: webapp/channels/src/components/post_priority/post_priority_picker_overlay.tsx [hunk 6]
```diff
@@ -104,6 +101,7 @@ function PostPriorityPickerOverlay({
                     className={classNames({control: true, active: pickerOpen})}
                     disabled={disabled}
                     type='button'
+                    aria-label={messagePriority}
```
Comment (saturninoabril) [inline webapp/channels/src/components/post_priority/post_priority_picker_overlay.tsx:104]: This fixes the issue in the screenshot.
Screenshot 2023-07-26 at 4 57 59 PM

File: webapp/channels/src/components/sidebar/__snapshots__/invite_members_button.test.tsx.snap [hunk 7]
```diff
@@ -68,17 +68,17 @@ exports[`components/sidebar/invite_members_button should match snapshot 1`] = `
...
-            id="introTextInvite"
+            id="inviteMembersButton"
             modalId="invitation"
             onClick={[Function]}
           >
             <button
-              aria-label="Invite Users dialog"
+              aria-label="Invite Members dialog"
               className="style--none intro-links color--link cursor--pointer"
-              id="introTextInvite"
```
Comment (saturninoabril) [inline webapp/channels/src/components/sidebar/__snapshots__/invite_members_button.test.tsx.snap:78]: This fixes the issue of duplicate ID.
Screenshot 2023-07-26 at 5 02 27 PM

File: webapp/channels/src/components/sidebar/__snapshots__/invite_members_button.test.tsx.snap [hunk 8]
```diff
@@ -93,7 +93,7 @@ exports[`components/sidebar/invite_members_button should match snapshot 1`] = `
                     Invite Members
                   </span>
                 </FormattedMessage>
-              </li>
```
Comment (saturninoabril) [inline webapp/channels/src/components/sidebar/__snapshots__/invite_members_button.test.tsx.snap:96]: This and the next one for "Add Channels" fixes the issue in the screenshot.

**Comment by saturninoabril** [comment]
@yasserfaraazkhan Was pulled out on other things but that's my plan too.
- added before fix in the description; and
- added comments to each changes and issues being addressed.

**Comment by saturninoabril** [comment]
@amyblais I'm adding this to 8.1 milestone to address customer issues on Accessibility.

File: webapp/channels/src/components/sidebar/sidebar_category_header.tsx [hunk 9]
```diff
@@ -40,6 +40,14 @@ type Props = StaticProps & {
 }
 
 export const SidebarCategoryHeader = React.forwardRef((props: Props, ref?: React.Ref<HTMLButtonElement>) => {
+    const {dragHandleProps} = props;
+
+    // (Accessibility) Ensures interactive controls are not nested as they are not always announced
+    // by screen readers or can cause focus problems for assistive technologies.
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    delete dragHandleProps?.role;
```
Comment (saturninoabril) [inline webapp/channels/src/components/sidebar/sidebar_category_header.tsx:49]: This fixes the issue in the screenshot.
Screenshot 2023-07-26 at 5 52 28 PM

Comment (M-ZubairAhmed) [inline webapp/channels/src/components/sidebar/sidebar_category_header.tsx:49]: nit: to remove eslint supressor

```
if (dragHandleProps && dragHandleProps.role) {
        Reflect.deleteProperty(dragHandleProps, 'role');
    }
    
```

**Review by M-ZubairAhmed — APPROVED** [review]

File: webapp/channels/src/components/sidebar/sidebar_category_header.tsx (hunk 9 above)
Comment (saturninoabril) [reply webapp/channels/src/components/sidebar/sidebar_category_header.tsx]: Thanks Zubair, applied and done.

**Comment by saturninoabril** [comment]
@asaadmahmood Once all good in E2E,I'm merging this so it can get into 8.1 but let me know if you have any comment.

**Pull request merged**